core/src/main/java/io/questdb/cairo/wal/CopyWalSegmentUtils.java

```diff
@@ -212,11 +221,14 @@ private static boolean copyTimestampFile(
         }
         long size = rowCount << 4;
         long srcDataTimestampAddr = TableUtils.mapRW(ff, primaryFd, size, MEMORY_TAG);
-        Vect.flattenIndex(srcDataTimestampAddr, rowCount);
-        if (commitMode != CommitMode.NOSYNC) {
-            ff.msync(srcDataTimestampAddr, size, commitMode == CommitMode.ASYNC);
+        try {
```

Comment:
Could you also check `copyVarSizeFiles` here? I don't understand why we're not doing munmap in the `finally` block:
```java
            // All in memory calls, no need to unmap in finally
            ff.munmap(newAuxMemAddr, newAuxMemSize, MEMORY_TAG);
```